From 5e1d0482919405695df558874153f00d6991430c Mon Sep 17 00:00:00 2001
From: Iain Lane <iain.lane@canonical.com>
Date: Thu, 28 Jan 2016 12:34:35 +0000
Subject: [PATCH 1/2] Hide GOA sources on Unity

(Backported to gnome-3-18, use the patches on the bug for 3.20)

https://bugzilla.gnome.org/show_bug.cgi?id=761259

---
 src/gcal-manager.c       | 14 ++++++++++----
 src/gcal-source-dialog.c |  9 +++++++++
 2 files changed, 19 insertions(+), 4 deletions(-)

Index: b/src/gcal-manager.c
===================================================================
--- a/src/gcal-manager.c
+++ b/src/gcal-manager.c
@@ -21,6 +21,7 @@
 #include "gcal-utils.h"
 
 #include <libedataserverui/libedataserverui.h>
+#include <string.h>
 
 typedef struct
 {
@@ -790,6 +791,7 @@
   GList *sources, *l;
   GError *error = NULL;
   ESourceCredentialsProvider *credentials_provider;
+  const gchar *desktop;
 
   G_OBJECT_CLASS (gcal_manager_parent_class)->constructed (object);
 
@@ -801,10 +803,14 @@
   manager->clients = g_hash_table_new_full ((GHashFunc) e_source_hash, (GEqualFunc) e_source_equal,
                                          g_object_unref, (GDestroyNotify) free_unit_data);
 
-  /* load GOA client */
-  goa_client_new (NULL, // we won't really cancel it
-                  (GAsyncReadyCallback) gcal_manager_client_ready_cb,
-                  object);
+  desktop = g_getenv ("XDG_CURRENT_DESKTOP");
+  if (!desktop || !strstr (desktop, "Unity"))
+    {
+      /* load GOA client */
+      goa_client_new (NULL, // we won't really cancel it
+                      (GAsyncReadyCallback) gcal_manager_client_ready_cb,
+                      object);
+    }
 
   /* reading sources and schedule its connecting */
   manager->source_registry = e_source_registry_new_sync (NULL, &error);
Index: b/src/gcal-source-dialog.c
===================================================================
--- a/src/gcal-source-dialog.c
+++ b/src/gcal-source-dialog.c
@@ -38,6 +38,7 @@
   GtkWidget          *main_scrolledwindow;
   GtkWidget          *name_entry;
   GtkWidget          *notebook;
+  GtkWidget          *online_account_main_grid;
   GtkWidget          *remove_button;
   GtkWidget          *stack;
   GtkWidget          *web_source_grid;
@@ -1788,11 +1789,18 @@
   GcalSourceDialog *self;
   GtkBuilder *builder;
   GMenuModel *menu;
+  const gchar *desktop;
 
   self = GCAL_SOURCE_DIALOG (object);
 
   G_OBJECT_CLASS (gcal_source_dialog_parent_class)->constructed (object);
 
+  desktop = g_getenv ("XDG_CURRENT_DESKTOP");
+  if (desktop && strstr (desktop, "Unity"))
+    {
+      gtk_widget_hide (self->online_account_main_grid);
+    }
+
   /* widget responses */
   gtk_dialog_set_default_response (GTK_DIALOG (object), GTK_RESPONSE_CANCEL);
 
@@ -1873,6 +1881,7 @@
   gtk_widget_class_bind_template_child (widget_class, GcalSourceDialog, notification);
   gtk_widget_class_bind_template_child (widget_class, GcalSourceDialog, notification_label);
   gtk_widget_class_bind_template_child (widget_class, GcalSourceDialog, online_accounts_listbox);
+  gtk_widget_class_bind_template_child (widget_class, GcalSourceDialog, online_account_main_grid);
   gtk_widget_class_bind_template_child (widget_class, GcalSourceDialog, owncloud_stub_row);
   gtk_widget_class_bind_template_child (widget_class, GcalSourceDialog, remove_button);
   gtk_widget_class_bind_template_child (widget_class, GcalSourceDialog, settings_button);
